src/data_structures/fenwick_tree.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/**
+ * Fenwick Tree (Binary Indexed Tree)
+ */
+function FenwickTree(length) {
+  this._elements = new Array(length + 1);
+  for (var i = 0; i < this._elements.length ; i++)
+    this._elements[i] = 0;
+}
+
+/**
+ * Adds value to the array at specified index in O(log n)
+ */
+FenwickTree.prototype.adjust = function (index, value) {
+  for (; index < this._elements.length ; index += (index&-index))
```

Comment:
The bit operations are easy to understand, but the rationale behind them would require explaining how the "tree" is represented inside the array. I don't think it would fit well inside the code. Is it ok to link a blog post?